### ISSUE TITLE
Precompile: 0x02 SHA2-256

### ISSUE THREAD

**Issue by ChihChengLiang** [privacy-scaling-explorations/zkevm-specs#319]

**Comment by ed255** [comment]
There's a sha256 circuit implementation in the halo2 repository (from zcash) which we should probably use for this precompile: https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_gadgets/src/sha256.rs

**Comment by CPerezz** [comment]
Agree! Let's make sure anyway that the sponge config parameters are indeed the same ones as we do need and this kind of stuff!

**Comment by Brechtpd** [comment]
Will plug my implementation as a possible candidate as well: https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/756

Should be significantly faster than the halo2 implementation and I think the implementation is also simpler (more straightforward implementation on bits), however at the cost of needing more columns (~130) (though I'm not sure how the implementations compare based on total cells required/hash). One other nice benefit is that my implementation already uses the same lookup API for passing data around as keccak (RLC encoded input and hash) which might be useful.
